File: apps/api/internal/domain/entities/project.go
package entities

import "time"

// ProjectStatus is the lifecycle of a project (PRD §3.1.1).
type ProjectStatus string

const (
	ProjectActive   ProjectStatus = "ACTIVE"
	ProjectArchived ProjectStatus = "ARCHIVED"
)

// Project is a business area owned by a tenant (PRD §3.1.1).
// A tenant owns many projects; each project owns many workflows.
type Project struct {
	ID        string
	TenantID  string
	Name      string
	Slug      string
	Status    ProjectStatus
	CreatedAt time.Time
	UpdatedAt time.Time
}

File: packages/schemas/project.ts
import { z } from "zod"

export const projectStatuses = ["ACTIVE", "ARCHIVED"] as const

export const projectResponseSchema = z.object({
  id: z.string(),
  tenantId: z.string(),
  name: z.string(),
  slug: z.string(),
  status: z.enum(projectStatuses),
  createdAt: z.string(),
  updatedAt: z.string(),
})

export const projectListResponseSchema = z.array(projectResponseSchema)

export type ProjectResponseSchemaProps = z.infer<typeof projectResponseSchema>
export type ProjectListResponseSchemaProps = z.infer<
  typeof projectListResponseSchema
>
